src/charm.py

```diff
@@ -392,6 +393,8 @@ def _disable_enable_restart(self, event: RunWithLock) -> None:
         self.snap.disable_enable()
         self.snap.start_snap_service()
 
+        time.sleep(20.0)
+
```

Comment:
question: Is this specifically related to storage being added?

Reply:
No, no, this is in Kafka. When testing the upgrades under load I noticed that rolling restart units too quick (without timeouts) could get the clients a bit in trouble. Basically, a consumer would try to refresh the metadata and use a new broker to fetch data, but that broken may have meanwhile gone offline, screwing a bit the process. tens of seconds seemed good enough to get things a bit more stable and robust to allow for broker rejoining, catching up and also clients transitioning from one broker to another

Reply:
I think what Raul meant here, is that `disable_enable_restart` is a hack call we do to get around snaps not loading the new storages until it's 're-enabled'. I'm not sure that's needed for upgrades?